### PR TITLE
Feature[CON-124+125]: Hero Banner Fifty Fifty and Full Width Tracking

### PR DESCRIPTION
**Pull request by hassanraza0054**
Relates to https://github.com/aesop/aesop-web-ui/pull/3870 in web-ui

### CODE REVIEW

File: src/components/HeroBanner/HeroBanner.types.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { ReactElement, ReactNode } from 'react';
+import { ReactElement, ReactNode, ForwardedRef } from 'react';
```
Comment (beau-charman) [inline src/components/HeroBanner/HeroBanner.types.ts:1]: where is `ForwardedRef` being used?

File: src/components/HeroBanner/components/HeroBannerMedia/HeroBannerMedia.types.ts [hunk 2]
```diff
@@ -9,8 +9,9 @@ type HeroBannerMediaProps = {
   containMedia?: HeroBannerMediaAnchorPoints;
   foregroundImage?: ReactElement;
   foregroundImageLink?: {
-    url?: string;
+    onClick?: () => void;
     title?: string;
+    url?: string;
```
Comment (beau-charman) [inline src/components/HeroBanner/components/HeroBannerMedia/HeroBannerMedia.types.ts:14]: Thank you for reordering too :)

File: src/components/SubNav/SubNav.tsx [hunk 3]
```diff
@@ -4,14 +4,23 @@ import { useThemeContext } from '~/contexts';
 import { useWindowHasResized } from '~/customHooks';
 import { isViewport } from '~/utils/viewport';
 import { List } from '~/components/List';
-import { Select } from '~/components/Select/index.ts';
-import { Heading } from '~/components/Heading/index.ts';
+import { Select } from '~/components/Select/index';
+import { Heading } from '~/components/Heading/index';
```
Comment (beau-charman) [inline src/components/SubNav/SubNav.tsx:8]: Thank you!

File: src/components/SubNav/SubNav.tsx [hunk 4]
```diff
@@ -41,7 +50,10 @@ const SubNav = forwardRef<HTMLElement, SubNavProps>(function SubNavRef(
         <Select
           isBlock={isSelect}
           name={id}
-          onChange={handleOnChange}
+          onChange={(e) => {
```
Comment (beau-charman) [inline src/components/SubNav/SubNav.tsx:53]: Sorry to be a pain, could when keep the event object named `event`? Thank you

File: src/components/SubNav/SubNav.types.ts [hunk 5]
```diff
@@ -28,6 +28,7 @@ type SubNavProps = {
   heading?: string;
   headingClassName?: string;
   theme?: Themes;
+  onSelectCallback?: (e) => void;
```
Comment (beau-charman) [inline src/components/SubNav/SubNav.types.ts:31]: Not sure why you're not getting a type error here, but either specify the event type, or omit it. 

For example: 

```
(event: React.ChangeEvent<HTMLSelectElement>) => void;
(event: MouseEvent) => void;
() => void;
```

File: src/components/HeroBanner/HeroBanner.types.ts (hunk 1 above)
Comment (hassanraza0054) [reply src/components/HeroBanner/HeroBanner.types.ts:1]: Ah good catch. Must have been leftover when I was trying to use the impression function callback in gel. Turned out I didnt need to use it for this, so I thought I removed it but must've missed it. Adjusting now

File: src/components/SubNav/SubNav.types.ts (hunk 5 above)
Comment (hassanraza0054) [reply src/components/SubNav/SubNav.types.ts]: Done! Ty

**Review by MkMan — APPROVED** [review]

**Review by zack76 — APPROVED** [review]

**Review by beau-charman — APPROVED** [review]

**Pull request merged**